test/ICSharpCode.SharpZipLib.Tests/Zip/StreamHandling.cs

```diff
@@ -556,5 +564,43 @@ public void ShouldThrowDescriptiveExceptionOnUncompressedDescriptorEntry()
 				}
 			}
 		}
+		
+		[Test]
+		[Category("Zip")]
+		public void IteratingOverEntriesInDirectUpdatedArchive()
+		{
+			using (var tempFile = new Utils.TempFile())
+			{
+				var fi = new FileInfo(tempFile.Filename);
+				
+				using (var zf = ZipFile.Create(tempFile.Filename))
+				{
+					zf.BeginUpdate();
+					// Add a "large" file
+					zf.Add(new MemoryDataSource(new byte[1024]), "FirstFile", CompressionMethod.Stored);
```

Comment:
Currently only tests skipping null-bytes (since the first file content is used an array of `default(byte)` (`0`).
Add some non-zero data to the end of the data source to additionally test non-null seeking:
https://github.com/icsharpcode/SharpZipLib/blob/c6ccef30ed603897e665d4d65afae73b257dfb70/src/ICSharpCode.SharpZipLib/Zip/ZipInputStream.cs#L324-L328